### PR TITLE
Multipart post attachments

### PR DESCRIPTION
**Pull request by saneshark**
- Improves client response errors to include the MHV developer message for test and dev environments to assist debugging.
- Improves validations between: MessageDraft, Message, and Message that is a reply
- Implements validations on uploads for secure messaging
- Implements submitting multiple attachments for both New Message and Reply 
- Updates documentation to reflect how to POST and GET attachments

### CODE REVIEW

**Comment by mphprogrammer** [comment]
LGTM!

File: app/models/message.rb [hunk 1]
```diff
@@ -3,9 +3,19 @@
 
 # Message model
 class Message < Common::Base
+  MEGABYTE = 1024.0 * 1024.0
+  MAX_TOTAL_FILE_SIZE = 6.0
```
Comment (carlmjohnson) [inline app/models/message.rb]: Instead of dividing by MB later on, why not just define `MAX_TOTAL_FILE_SIZE = 6.0 * MEGABYTE`?

Comment (carlmjohnson) [inline app/models/message.rb]: TIL. Ruby has a method for everything.

Comment (saneshark) [inline app/models/message.rb]: I will look into making these changes retroactively in future PR since this got merged in already.

**Pull request merged**